src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -285,6 +285,7 @@ export default class QueryEditorUI extends Component<Props, State> {
       indexPattern,
       position,
       services: this.services,
+      dataset: this.queryString.getQuery().dataset,
```

Comment:
Do we need to pass `dataset` in here if `services` is already being passed in? It's already included in `services` if you use `services.data.query.queryString.getQuery().dataset`.

Reply:
i guess a lot of the parameters can be removed if we can get querystring from services, let me see about that